test/testdata/infer/generics/isa_with_type_member.rb

```diff
@@ -0,0 +1,99 @@
+# typed: true
+
+class A
+  extend T::Generic
+  X = type_member
+end
+
+class B < A
+  extend T::Generic
+  X = type_member(fixed: String)
+end
+
+class Test
+  extend T::Sig
+
+  sig {params(x: A[T.untyped]).void}
+  def test1(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `B`
+    when A
+      T.reveal_type(x) # error: Revealed type: `A[T.untyped]`
+    end
+  end
+
+  sig {params(x: A[String]).void}
+  def test2(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `B`
+    when A
+      T.reveal_type(x) # error: Revealed type: `A[String]`
+    end
+  end
+
+  sig {params(x: T.nilable(A[Integer])).void}
+  def test3(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `T.all(B, A[Integer])`
+    when A
+      T.reveal_type(x) # error: Revealed type: `A[Integer]`
+    end
+  end
+
+  sig {params(x: T.nilable(A[String])).void}
+  def test4(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `B`
+    when A
+      T.reveal_type(x) # error: Revealed type: `A[String]`
+    end
+  end
```

Comment:
This looks like a duplicate of `test2`?

Reply:
It's not quite, as the argument is `T.nilable`. This tripped a few cases in `lub` that the other didn't.